### ISSUE TITLE
Add options to set text outline color and width

### ISSUE THREAD

**Issue by sandboiii** [Raymo111/i3lock-color#147]
## This issue is a....

<pre>
[ ] Bug
[ x ] Other kind of issue (Please describe in detail)
</pre>
Hi, I'm very happy with i3lock-color. I've setup background blur on lock, but there is a problem that sometimes text is not visible when text color is similar to background's color. It would be great if I could set a border to the text, so it would be always visible on any background.

## Current Behavior
No way to set text border

## Expected Behavior
Some options which would allow to set borders for text like `--timeborder`, `--dateborder`

## Environment
Output of `i3lock --version`:
<pre>
i3lock version: aur-eabfff3
</pre>

Where'd you get i3lock-color from?

<pre>
[ x ] AUR package (which one?) i3lock-color-git
[ ] Built from source yourself
[ ] Other (Please describe in detail)
</pre>

**Comment by PandorasFox** [comment]
Unless Cairo supports this kind of behaviour natively, this won't happen.

**Comment by sandboiii** [comment]
@Raymo111 can you reopen this issue as enhancement? I would really want to see this implemented some day in the fututre

**Comment by Raymo111** [comment]
@sandboiii Absolutely. I'll try to implement this, but no guarantees :)

**Comment by sandboiii** [comment]
Great! Thank you very much

**Comment by Raymo111** [comment]
@sandboiii by border what do you mean? Can you show me an example?

**Comment by sandboiii** [comment]
> @sandboiii by border what do you mean? Can you show me an example?

Maybe I used the wrong term. I'm talking about being able to add a stroke or an outline for text. Like in examples:  
![outline](https://user-images.githubusercontent.com/17468894/82693559-6e58c800-9c6a-11ea-8c24-158370ec13ce.png)  
![stroke](https://user-images.githubusercontent.com/17468894/82693735-b24bcd00-9c6a-11ea-826d-3356a518add1.png)  

The reason why I need it is that I have setup background blur for lock screen. My system is almost entirely colored dark and I have contrasting white text for locker and time. But, when system locks on white background like when I use browser, then locker text and time is not visible. And to solve this problem I would like to see option to add text outline.

**Comment by Raymo111** [comment]
@sandboiii ahh, I see. I'd have to see if that's even possible! i.e. even in Google Docs there's no option for that. I wonder if Cairo supports this.

**Comment by Raymo111** [comment]
@sandboiii Seems like it _is_ possible. I just need to use
```
cairo_text_path (cr, "void");
cairo_set_source_rgb (cr, 0.5, 0.5, 1);
cairo_fill_preserve (cr);
cairo_set_source_rgb (cr, 0, 0, 0);
cairo_set_line_width (cr, 2.56);
cairo_stroke (cr);
```
Options for outline width and color should be easy enough to implement, but it will be a pain for every single text element. I'll start with one outline width and color for everything and go from there.

**Comment by sandboiii** [comment]
@Raymo111 this is great news! I was thinking that if there would be no stroke, adding text shadow is also a working trick, but not as good as clear outline. The example:
![whiteletteringwhitebackground](https://user-images.githubusercontent.com/17468894/82695709-4c614480-9c6e-11ea-8c00-efe6e0837cc3.jpg)

**Comment by Raymo111** [comment]
Yeah I'm not gonna do shadow, I'll stick with outline :)

**Comment by Qwerty-Space** [comment]
I know it's not been long, but do you have an update on this?

**Comment by Raymo111** [comment]
@Qwerty-Space sorry, I've been super busy lately... I'll give this a shot soon. Hang in there :)

**Comment by Qwerty-Space** [comment]
Any update?  :-)

**Comment by julio-b** [comment]
If anyone still needs this, i implemented outlines in #198.

**Comment by sandboiii** [comment]
@julio-b many thanks to your work!
